WalletWasabi/Blockchain/Keys/WalletGenerator.cs

```diff
@@ -40,6 +41,21 @@ public WalletGenerator(string walletsDir, Network network)
 		return (km, mnemonic);
 	}
 
+	public (KeyManager,  Share[]) GenerateWallet(string walletName, string password, Share[]? shares = null)
+	{
+		string walletFilePath = GetWalletFilePath(walletName, WalletsDir);
+
+		// Here we are not letting anything that will be autocorrected later. We need to generate the wallet exactly with the entered password because of compatibility.
+		PasswordHelper.Guard(password);
+
+		var km = shares is null
+			? KeyManager.CreateNew(out shares, password, Network)
+			: KeyManager.CreateNew(shares, password, Network);
```

Comment:
```suggestion
		PasswordHelper.Guard(password);

		shares ??= Shamir.Generate(
			DefaultShamirThreshold,
			DefaultShamirShares,
			RandomUtils.GetBytes(32)).ToArray();

		var km = KeyManager.CreateNew(shares[..DefaultShamirThreshold], password, Network)
		
```

Reply:
shares[..DefaultShamirThreshold] will not work if shares was actually provided

Reply:
https://github.com/WalletWasabi/WalletWasabi/pull/13800/commits/277c0ab1f2767be44ea04695c9e8d78a0a8f712e